portfolio/src/main/java/com/whz/portfolio/resource/PortfolioResource.java

```diff
@@ -27,83 +27,100 @@
+	 */
+	public Completes<Response> handleGetPortfolio(String id) {
+		return portfolioQueries.portfolioOf(id)
+				.andThenTo(PortfolioData.empty(),
+						data -> Completes.withSuccess(
+								Response.of(Ok, headers(of(ContentType, "application/json")), serialized(data))))
+				.otherwise(noData -> Response.of(NotFound, "/portfolio/" + id));
+	}
 
-  // POST
-  public Completes<Response> handlePost(PortfolioData data) {
-    return Portfolio.defineWith(stage, data.owner)
-        .andThenTo(
-            state ->
-                Completes.withSuccess(
-                    Response.of(
-                        Created,
-                        headers(of(Location, location(state.id)))
-                            .and(of(ContentType, "application/json")),
-                        serialized(PortfolioData.from(state)))));
-  }
+	/**
+	 * Create a new portfolio.
+	 * 
+	 * @param owner
+	 * @return JSON representation of the created PortfolioData
+	 */
+	public Completes<Response> handleCreatePortfolio(String owner) {
+		return Portfolio.defineWith(stage, owner)
+				.andThenTo(state -> Completes.withSuccess(Response.of(Created,
+						headers(of(Location, "/portfolio/" + state.id)).and(of(ContentType, "application/json")),
+						serialized(PortfolioData.from(state)))));
+	}
 
-  // PUT
-  public Completes<Response> handlePut(PortfolioData data) {
-    return Completes.withSuccess(
-        Response.of(Ok, headers(of(Location, location(data.id))), "Updated"));
-  }
+	/**
+	 * Add a stock to an existing portfolio. The value of the purchase gets published to the account.
+	 * 
+	 * @param id
+	 * @param data
+	 * @return JSON representation of the associated PortfolioData
+	 */
+	public Completes<Response> handleAcquireStock(String id, AcquireStockData data) {
+		return resolve(id).andThenTo(portfolio -> {
+			StockQuoteData stockQuoteData = QuotesCache.INSTANCE.get(data.symbol);
+			StockAcquiredPublisher.INSTANCE.send(data.amount * stockQuoteData.regularMarketPrice);
+			return portfolio.stockAcquired(data.symbol, stockQuoteData.regularMarketTime, data.amount,
+					stockQuoteData.regularMarketPrice);
+		}).andThenTo(state -> Completes.withSuccess(
+				Response.of(Ok, headers(of(ContentType, "application/json")), serialized(PortfolioData.from(state)))))
+				.otherwise(noData -> Response.of(NotFound, "/portfolio/" + id));
+	}
 
-  // DELETE
-  public Completes<Response> handleDelete(String id) {
-    return Completes.withSuccess(Response.of(Ok, "Deleted"));
-  }
+	/**
+	 * Retrieve a specific stock.
+	 * 
+	 * @param symbol
+	 * @return JSON representation of StockQuoteData
+	 */
+	public Completes<Response> handleGetStock(String symbol) {
+		StockQuoteData result = StockQuoteSubscriber.INSTANCE.get(symbol);
+		if (result == null) {
+			Response.of(NotFound, "/stocks/" + symbol);
+		}
+		return Completes.withSuccess(Response.of(Ok, headers(of(ContentType, "application/json")), serialized(result)));
+	}
```

Comment:
Why didn't we use a query to get the data from the cache? Definitely, I was wondering about line 96?